pkg/api/handlers/projectinvitation.go

```diff
@@ -0,0 +1,332 @@
+package handlers
+
+import (
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ProjectInvitationHandler struct{}
+
+func NewProjectInvitationHandler() *ProjectInvitationHandler {
+	return &ProjectInvitationHandler{}
+}
+
+func (h *ProjectInvitationHandler) CreateInvitationForProject(req api.Context) error {
+	thread, err := getProjectThread(req)
+	if err != nil {
+		return err
+	}
+
+	if !thread.Spec.Project {
+		return types.NewErrBadRequest("only projects can have invitations")
+	}
+
+	if thread.Spec.UserID != req.User.GetUID() {
```

Comment:
Should it be possible for an admin to invite someone to anyone else's project?

Reply:
Yes, fixed